dht/i_partitioner.cc

```diff
@@ -191,7 +205,7 @@ selective_token_range_sharder::next() {
         auto intersection = _range.intersection(std::move(candidate), dht::token_comparator());
```

Comment:
We might be able to make it work on a lower level (and so on more cases) by changing the token compare functions, so that intersection() returns a disengaged optional.

Although, interval will not be able to distinguish between an open interval and min/max tokens. So it's probably not workable.

Reply:
I don't think it can be done while keeping the interval classes generic.

Reply:
Agree, if we remove before_all/after_all _and_ specialize `interval` for integral ranges, _and_ communicate to `interval` that `token` is integral, we might pull it off. Quite a large task.

Reply:
Let's leave it as a future improvement, shall we?